tomb-wasm/src/mount.rs

```diff
@@ -437,7 +437,7 @@ impl WasmMount {
     /// * `Bucket is locked` - If the bucket is locked
     pub async fn ls(&mut self, path_segments: Array) -> TombResult<Array> {
         // Read the array as a Vec<String>
-        let path_segments = path_segments
+        let path_segments = &path_segments
```

Comment:
Another weird one, and this will actually force an additional copy of the internal strings later on when its joined where ownership of the internal state could just be consumed.